src/main/java/com/bobocode/hoverla/bring/context/BeanConfigClassValidator.java

```diff
@@ -0,0 +1,122 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.annotation.Bean;
+import com.bobocode.hoverla.bring.annotation.Configuration;
+import com.bobocode.hoverla.bring.exception.BeanConfigValidationException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Validates classes annotated with {@link Configuration @Configuration} and
+ * their methods marked with {@link Bean @Bean} that were scanned by {@link BeanScanner}.
+ *
+ * <p>Validation ensures:
+ * <ol>
+ *   <li>Valid type of a {@link Class} config object - it should be plain public class.
+ *   <br>See {@link BeanConfigClassValidator#checkClassModifiers(Class, List)} and {@link BeanConfigClassValidator#checkConfigIsPlainClass(Class, List)}</li>
+ *   <li>Valid bean methods - they should be public and non-static.
+ *   <br>See {@link BeanConfigClassValidator#checkMethodModifiers(Method, List)}</li>
+ *   <li>Valid constructor - bean config class should have a public non-args constructor.
+ *   <br>See {@link BeanConfigClassValidator#checkNoArgsConstructorExists(Class, List)}
+ * </ol>
+ *
+ * @see Bean
+ * @see Configuration
+ * @see BeanScanner
+ */
+@Slf4j
+class BeanConfigClassValidator {
```

Comment:
Why is this class package-private?

Reply:
As it's not used in any other places. May return public for consistency, but we should be back to encapsulation in module story

Reply:
Makes sense, let's keep it public for consistency for now. We will get back to it as we start with the module story

Reply:
Fixed